mender-conversion-tool.sh

```diff
@@ -0,0 +1,688 @@
+        device    - target system device where the script writes to
+        rootsize  - size in sectors (sector size 512 bytes) of the root 
+                    filesystem
+        size-data - size of data partition in MiB; default value 128MiB
+        platform  - target device identification used to build .sdimg name
+
+        Note: 'rootsize' can be found as an output from 'prepare_image' command
+
+Examples:
+
+    To shrink the existing embedded image:
+
+        ./mender-conversion-tool.sh prepare_image --image IMAGE_NAME
+
+        Output: Root filesystem size (sectors): 4521984
+
+    To prepare SD card:
+
+        ./mender-conversion-tool.sh make_sdcard --image REWORKED_IMAGE_NAME
+                --device /dev/sd[a-z] or /dev/mmcblk[0-9]
+                --rootsize ROOTSIZE --size-data 128
+
+        Output: ready to use SD card which can be mounted directly in a target
+                device (BeagleBone Black, Raspberry, etc.)
+
+    To prepare .sdimg file:
+
+        ./mender-conversion-tool.sh make_sdimg --image REWORKED_IMAGE_NAME
+                --rootsize ROOTSIZE --size-data 128 --platform beaglebone
+
+	Output: ready to use .sdimg file which can be used to flash SD card
+
+EOF
+}
+
+if [ $# -eq 0 ]; then
+  show_help
+  exit 1
+fi
+
+BASE_DIR=${PWD}
+
+# Default sector size
+sector_size=
+# Erase block size 8MiB
+erase_block=8388608
+# Boot partition start in sectors (512 bytes per sector).
+pboot_start=
+# Default 'boot' partition size in sectors: 16MiB (16777216 bytes)
+pboot_size=
+# Default 'data' partition size in MiB.
+data_size=128
+# Data partition size in sectors.
+pdata_size=
+# beaglebone: 4521984
+prootfs_size=
+
+target_dev=
+platform=
+image=
+number_of_parts=
+loopsdimg=
+
+DEV_CARD=1
+DEV_IMG=2
+
+declare -a part_names=("boot" "primary" "secondary" "data")
+declare -a mappings
+
+get_part_number_from_device() {
+    case "$1" in
+        /dev/*[0-9]p[1-9])
+            echo ${1##*[0-9]p}
+            ;;
+        /dev/[sh]d[a-z][1-9])
+            echo ${1##*d[a-z]}
+            ;;
+        ubi[0-9]_[0-9])
+            echo ${1##*[0-9]_}
+            ;;
+        [a-z]*\.sdimg[1-9])
+            echo ${1##*\.sdimg}
+            ;;
+        /dev/mapper/*[0-9]p[1-9])
+            echo ${1##*[0-9]p}
+            ;;
+        *)
+            echo "Could not determine partition number from $1"
+            exit 1
+            ;;
+    esac
+}
+
+unmount_partitions() {
+  echo "1. Check if device is mounted..."
+  is_mounted=`grep ${target_dev} /proc/self/mounts | wc -l`
+  if [ ${is_mounted} -ne 0 ]; then
+    echo $sudoPW | sudo -S umount ${target_dev}?*
+  fi
+}
+
+erase_filesystem() {
+  echo "2. Erase filesystem..."
+  echo $sudoPW | sudo -S wipefs --all --force ${target_dev}?*
+  echo $sudoPW | sudo -S dd if=/dev/zero of=${target_dev} bs=1M count=100
+}
+
+process_source_image() {
+  local image_name=$1
+  subname=${image_name:0:8}
+
+  image_data="$(echo $sudoPW | sudo -S fdisk -u -l ${image_name})"
```

Comment:
This is an example where you probably do not need `sudo`

Reply:
Right.